src/app/app.component.ts

```diff
@@ -19,7 +22,7 @@ export class AppComponent implements OnInit {
 		test: new FormControl(),
 	});
 
-	constructor(private optionService: OptionService) {}
+	constructor(private optionService: OptionService, private dialog: MatDialog) {}
```

Comment:
create modal.module with custom modal.service that will use MatDialog and have method open